src/server/pps/server/master.go

```diff
@@ -42,8 +44,31 @@ var (
 	falseVal bool  // used to delete RCs in deletePipelineResources and restartPipeline()
 )
 
+type pipelineKey string
+
+func toKey(p *pps.Pipeline) pipelineKey {
```

Comment:
isn't this the same as p.String()?

Reply:
It _is_, but only accidentally: `Pipeline.String()` returns a human-legible string, while `toKey` needs to be a uniquely-identifying key.

Reply:
isn't Pipeline.String() also always going to be a unique name?